src/starknetCli.ts

```diff
@@ -288,26 +366,65 @@ export async function runStarknetCallOrInvoke(
     throw e;
   }
 
+  const cmd: string = starkNetCallOrInvoke(abiPath, callOrInvoke, options, funcName, inputs);
+
+  return cmd;
+}
+
+function declareContract(filePath: string, option?: IDeclareOptions): string | undefined {
+  const network: string = option?.network ? `--network ${option.network}` : ``;
+
+  const options: Map<string, string | undefined> = new Map([['contract', filePath]]);
+
+  const multiOptions: string[] = [network];
+
+  const cmd: string = buildCairoCommand(options, CAIRO_CMD_DECLARE, multiOptions);
+  const result = callCairoCommand(cmd, CAIRO_CMD_DECLARE, false) as string;
   try {
-    execSync(
-      `${warpVenvPrefix} starknet ${callOrInvoke}  --address ${options.address} --abi ${abiPath} --function ${funcName} --network ${options.network} ${wallet} ${account} ${inputs}`,
-      { stdio: 'inherit' },
-    );
+    return processDeclareCLI(result, filePath);
   } catch {
-    logError(`starknet ${callOrInvoke} failed`);
+    logError('StarkNet declare failed');
   }
 }
 
-export function runStarknetDeclare(filePath: string) {
+export function runStarknetDeclare(filePath: string, options: IDeclareOptions) {
```

Comment:
What do you think about splitting up the options in this function and passing them individually to the declare function? Could be either way.

Reply:
I think using options would be better, it would be easier to add if there were any changes. And I guess that it is only taking 1 option, but for the rest of the commands using options seems to be the way its done, if we were to change declare, I think we should change the others.